azext_iot/iothub/commands_device_messaging.py

```diff
@@ -0,0 +1,221 @@
+    login: str = None,
+):
+    simulator_provider = DeviceMessagingProvider(
+        cmd=cmd, device_id=device_id, hub_name=hub_name, rg=resource_group_name, login=login
+    )
+    return simulator_provider.device_send_message(
+        data=data,
+        properties=properties,
+        msg_count=msg_count,
+        device_symmetric_key=device_symmetric_key,
+        certificate_file=certificate_file,
+        key_file=key_file,
+        passphrase=passphrase
+    )
+
+
+def iot_c2d_message_complete(
+    cmd,
+    device_id: str,
+    etag: str = None,
+    hub_name: str = None,
+    resource_group_name: str = None,
+    login: str = None,
+):
+    simulator_provider = DeviceMessagingProvider(
+        cmd=cmd, device_id=device_id, hub_name=hub_name, rg=resource_group_name, login=login
+    )
+    return simulator_provider.c2d_message_complete(
+        etag=etag
+    )
+
+
+def iot_c2d_message_reject(
+    cmd,
+    device_id: str,
+    etag: str = None,
+    hub_name: str = None,
+    resource_group_name: str = None,
+    login: str = None,
+):
+    simulator_provider = DeviceMessagingProvider(
+        cmd=cmd, device_id=device_id, hub_name=hub_name, rg=resource_group_name, login=login
+    )
+    return simulator_provider.c2d_message_reject(
+        etag=etag
+    )
+
+
+def iot_c2d_message_abandon(
+    cmd,
+    device_id: str,
+    etag: str = None,
+    hub_name: str = None,
+    resource_group_name: str = None,
+    login: str = None,
+):
+    simulator_provider = DeviceMessagingProvider(
+        cmd=cmd, device_id=device_id, hub_name=hub_name, rg=resource_group_name, login=login
+    )
+    return simulator_provider.c2d_message_abandon(
+        etag=etag
+    )
+
+
+def iot_c2d_message_receive(
+    cmd,
+    device_id: str,
+    lock_timeout: int = 60,
+    abandon: bool = False,
+    complete: bool = False,
+    reject: bool = False,
+    hub_name: str = None,
+    resource_group_name: str = None,
+    login: str = None,
+):
+    simulator_provider = DeviceMessagingProvider(
+        cmd=cmd, device_id=device_id, hub_name=hub_name, rg=resource_group_name, login=login
+    )
+    return simulator_provider.c2d_message_receive(
+        lock_timeout=lock_timeout, abandon=abandon, complete=complete, reject=reject
+    )
+
+
+def iot_c2d_message_send(
+    cmd,
+    device_id: str,
+    data: str = "Ping from Az CLI IoT Extension",
+    message_id: str = None,
+    correlation_id: str = None,
+    user_id: str = None,
+    content_encoding: str = "utf-8",
+    content_type: str = None,
+    expiry_time_utc=None,
+    properties: str = None,
+    ack=None,
+    wait_on_feedback=False,
+    yes: bool = False,
+    repair: bool = False,
+    hub_name: str = None,
+    resource_group_name: str = None,
+    login: str = None,
+    auth_type_dataplane: str = None
+):
+    from azext_iot.common.deps import ensure_uamqp
+    ensure_uamqp(cmd.cli_ctx.config, yes, repair)
+
+    simulator_provider = DeviceMessagingProvider(
```

Comment:
nit, not everything is "simulation" related,  sending a c2d message is not a device simulation activity, but acknowledging the message is.